bin/node/runtime/src/lib.rs

```diff
@@ -513,6 +513,11 @@ parameter_types! {
 		.get(DispatchClass::Normal)
 		.max_extrinsic.expect("Normal extrinsics have a weight limit configured; qed")
 		.saturating_sub(BlockExecutionWeight::get());
+	// Solution can occupy 90% of normal block size
+	pub MinerMaxLength: u32 = RuntimeBlockLength::get()
+		.max
+		.get(DispatchClass::Normal)
+		.saturating_mul(9) / 10;
```

Comment:
Perbill can be used to avoid the risk of saturation. 

Or make it `pub const MinerMaxLength ` and use normal arithmetic to make sure it fails at compilation.

Reply:
Took a weirdly long time to figure out that my type errors were because `Perbill::from_rational` is defined on arbitrary `N`, but https://github.com/paritytech/substrate/pull/8317/commits/b39dac102ffc5727df171190ac8fa69bc10d5be4 should solve this. I'll update in Polkadot as well.